Lifestraw

Maleclypse opened the pull request:
#### Summary

Summary: Content "Lifestraw"

#### Purpose of change

Lifestraws are becoming a fairly common survivalist tool.

#### Describe the solution

Since it uses a combination of charcoal filters and internal fibers to filter out both muck, parasites and disease. I modeled it after the Charcoal water purifer.  I made it a lower end version with less liters of water filtered per charge in order to respect game balance, since it's size is so much by smaller weight and volume.

#### Describe alternatives you've considered

Trying to utilize the have a drink feature from the vehicle kitchen rig.

#### Additional context

Maleclypse committed:
Merge pull request #1 from CleverRaven/master

Update

Maleclypse committed:
Merge pull request #2 from CleverRaven/master

Catching up to current version

Maleclypse committed:
Update tools.json

Added lifestraw

Maleclypse committed:
Update tools.json

corrected syntax

Maleclypse committed:
Update tools.json

changed color to one  i know is in game

Maleclypse committed:
Update tools.json

pulled hanging comma

snipercup commented:
How would one aquire a lifestraw? Will there be a crafting recipe?

ZhilkinSerg reviewed (changes requested):
You need to spawn new item somewhere - like in sporting goods or hunting stores. Otherwise it can only be spawned via debug menu.

DaviBones commented:
I recommend going a different direction with the balance, make it have a realistic number of uses (good for 1000-4000 liters depending on model, according to lifestraw.com) but balance it by making it quite a rare find (these would have been an item highly sought-after by looters), and non-craftable.

The best content contributions are the ones that satisfy both the game-balance crowd and the realism crowd simultaneously.

Maleclypse committed:
Update tools.json

Added spawns  in both hunting gear and survival goods.  If hunting gear is more likely to found in the wild rather than in stores I'll up droprate more.

Maleclypse committed:
Update tools.json

Max charrges and price

Maleclypse commented:
> I recommend going a different direction with the balance, make it have a realistic number of uses (good for 1000-4000 liters depending on model, according to lifestraw.com) but balance it by making it quite a rare find (these would have been an item highly sought-after by looters), and non-craftable.
> 
> The best content contributions are the ones that satisfy both the game-balance crowd and the realism crowd simultaneously.
> 
> EDIT: As far how to spawn it, I would start by adding it to the `tools_survival` item group in `data\json\itemgroups\tools.json`. There might be other item groups it would fit into nicely as well.

I added it to hunting gear and survival gear.  I've moved the number of units of water that it can clean to 1000 liters based on the $40 model I got my weight and volume from.  At 250ml per charge that means it needed 4k charges, but it can't be recharged.

Maleclypse commented:
> How would one aquire a lifestraw? Will there be a crafting recipe?

I believe the technology to manufacture them is probably lost to the survivor barring him finding a lifestraw factory.  So I've added them as a low spawn in hunting and survival gear as they will be pretty heavily scavenged by people who know what they are looking for.  If I learn how to do so I might make one a 50% chance spawn in LMOEs.  I'm not even sure that's possible yet at what I know so far.

Shodan14 commented:
> 
> 
> I recommend going a different direction with the balance, make it have a realistic number of uses (good for 1000-4000 liters depending on model, according to lifestraw.com) but balance it by making it quite a rare find (these would have been an item highly sought-after by looters), and non-craftable.
> 
> The best content contributions are the ones that satisfy both the game-balance crowd and the realism crowd simultaneously.
> 
> EDIT: As far how to spawn it, I would start by adding it to the `tools_survival` item group in `data\json\itemgroups\tools.json`. There might be other item groups it would fit into nicely as well.
It should also be slow to use, as the real-life version is.

Maleclypse commented:
> > I recommend going a different direction with the balance, make it have a realistic number of uses (good for 1000-4000 liters depending on model, according to lifestraw.com) but balance it by making it quite a rare find (these would have been an item highly sought-after by looters), and non-craftable.
> > The best content contributions are the ones that satisfy both the game-balance crowd and the realism crowd simultaneously.
> > EDIT: As far how to spawn it, I would start by adding it to the `tools_survival` item group in `data\json\itemgroups\tools.json`. There might be other item groups it would fit into nicely as well.
> 
> It should also be slow to use, as the real-life version is.

How would I edit the time it takes to use?

Night-Pryanik commented:
> How would I edit the time it takes to use?

It would require delving deep into the code. You could look at how fire drill firestarter iuse action is implemented.

Maleclypse committed:
Update tools.json

added moves cost of 80 seconds

Maleclypse commented:
> > How would I edit the time it takes to use?
> 
> It would require delving deep into the code. You could look at how fire drill firestarter iuse action is implemented.
I  looked at what you suggested and compared it to hiker reviews plus user manual instructions and came up with an estimate of 80 seconds to get 250 ML of fresh water.  Technically it would take less for successive drinks after that and possibly more on the front end.  Did I put that all together correctly?  Thank you for the advice and hopefully I did this correctly.

Maleclypse committed:
Update tools.json

thethunderhawk commented:
IRL lifestraws just use your mouth suction to pull the water through the filter, it has no pumping mechanism aside from that. You could theoretically draw filtered water into your mouth and then spit it into a container for storage, but that should take a long time.

ZhilkinSerg commented:
Gravity can do this for you.

Maleclypse commented:
> IRL lifestraws just use your mouth suction to pull the water through the filter, it has no pumping mechanism aside from that. You could theoretically draw filtered water into your mouth and then spit it into a container for storage, but would that water stay safe to drink long term after picking up the bacteria in your mouth?

I could try and change this to a "have drink" action tied with a purify water but honestly that is beyond my skill level.

anothersimulacrum commented on data/json/items/tools.json:
```diff
@@ -1162,6 +1162,26 @@
+    "to_hit": -3,
+    "bashing": 1,
+    "material": [ "cotton", "plastic" ],
+    "symbol": "i",
```
Could you add a look_like to this? It would help make #34230 a little easier.

Maleclypse replied:
Would I add looks_like "pipe" or looks_like "lifestraw"?

Maleclypse replied:
I can also add additional details to the description so that eventually if someone wants to build a specific tile for it it will be easier?

anothersimulacrum replied:
For `looks_like`, you use the id of the thing you want it to look like.

Maleclypse committed:
Update tools.json

anothersimulacrum commented on data/json/items/tools.json:
```diff
@@ -1162,6 +1162,27 @@
+    "bashing": 1,
+    "material": [ "cotton", "plastic" ],
+    "symbol": "i",
+    "looks_like": "pipe"
```
```suggestion
    "looks_like": "pipe",
```

Maleclypse committed:
Update data/json/items/tools.json

Co-Authored-By: anothersimulacrum <anothersimulacrum@gmail.com>

Maleclypse committed:
Update tools.json

Maleclypse committed:
Update tools.json

Maleclypse committed:
Update tools.json

Maleclypse commented:
When I look at the failed Travis test I'm seeing that it ran too long on android and therefore it failed.  How do I fix that if at all?

anothersimulacrum commented:
You don't, it's not your fault, so no need to worry about it.

Maleclypse commented:
> Gravity can do this for you.

I'm not sure if you are asking me to make a change here?

Maleclypse committed:
Update tools.json

Maleclypse commented:
> You need to spawn new item somewhere - like in sporting goods or hunting stores. Otherwise it can only be spawned via debug menu.

I've added spawn locations.  Are there any other changes needed for this to merge?

adamkad1 commented:
i though that was going to be the pool noodle
